src/runtime-rs/crates/hypervisor/src/qemu/inner.rs

```diff
@@ -61,6 +61,11 @@ impl QemuInner {
     pub(crate) async fn start_vm(&mut self, _timeout: i32) -> Result<()> {
         info!(sl!(), "Starting QEMU VM");
 
+        let netns = self.netns.clone();
+        if let Err(e) = do_add_netns(&netns) {
+            return Err(anyhow!("add netns failed with error {:?}", e));
+        };
+
```

Comment:
This works, however it puts the executing thread permanently into the custom netns.  This is a departure from existing practice of `resource` crate code which rigorously ensures that the netns is only entered for operations that require it and is left afterwards.  I believe we should stick to that in the qemu driver as well.

The question is how to do it.  I'd very much like to reuse the `resource` crate's `NetnsGuard` for this, however we apparently cannot just `use` it since the `resource` crate already depends on the `hypervisor` crate so that would create a cyclic dependency.  One solution could be to move `NetnsGuard` to a lower-level utitility crate which could be imported by both `resource` and `hypervisor`, what do you think?